docs/high-availability.md

```diff
@@ -0,0 +1,199 @@
+[/etc/init.d/kubelet](high-availability/init-kubelet) and [/etc/default/kubelet](high-availability/default-kubelet)
+scripts.
+
+If you are using monit, you should also install the monit daemon (```apt-get install monit```) and the [/etc/monit/conf.d/kubelet](high-availability/monit-kubelet) and
+[/etc/monit/conf.d/docker](high-availability/monit-docker) configs.
+
+On systemd systems you ```systemctl enable kubelet``` and ```systemctl enable docker```.
+
+
+## Establishing a redundant, reliable data storage layer.
+The central foundation of a highly available solution is a redundant, reliable storage layer.  The number one rule of high-availability is
+to protect the data.  Whatever else happens, whatever catches on fire, if you have the data, you can rebuild.  If you lose the data, you're
+done.
+
+Clustered etcd already replicates your storage to all master instances in your cluster.  This means that to lose data, all three nodes would need
+to have their physical (or virtual) disks fail at the same time.  The probability that this occurs is relatively low, so for many people
+running a replicated etcd cluster is likely reliable enough.  You can add additional reliability by increasing the
+size of the cluster from three to five nodes.  If that is still insufficient, you can add
+[even more redundancy to your storage layer](#even-more-reliable-storage).
+
+### Clustering etcd
+The full details of clustering etcd are beyond the scope of this document, lots of details are given on the
+[etcd clustering page](https://github.com/coreos/etcd/blob/master/Documentation/clustering.md).  This example wals through
+a simple cluster set up, using etcd's built in discovery to build our cluster.
+
+First, hit the etcd discovery service to create a new token:
+
+```sh
+curl https://discovery.etcd.io/new?size=3
+```
+
+On each node, copy the [etcd.manifest](high-availability/etcd.manifest) file into ```/etc/kubernetes/manifests/etcd.manifest```
+
+The kubelet on each node actively monitors the contents of that directory, and it will create an instance of the ```etcd```
+server from the definition of the pod specified in ```etcd.manifest```.
+
+Note that in ```etcd.manifest``` you should substitute the token URL you got above for ```${DISCOVERY_TOKEN}``` on all three machines,
+and you should substitute a different name (e.g. ```node-1```) for ${NODE_NAME} and the correct IP address
+for ```${NODE_IP}``` on each machine.
+
+
+#### Validating your cluster
+Once you copy this into all three nodes, you should have a clustered etcd set up.  You can validate with
+```
+etcdctl member list
+```
+
+and
+
+```
+etcdctl cluster-health
+```
+
+You can also validate that this is working with ```etcdctl set foo bar``` on one node, and ```etcd get foo```
+on a different node.
+
+### Even more reliable storage
+Of course, if you are interested in increased data reliability, there are further options (belts *and* suspenders, ftw!).
```

Comment:
maybe "there are further options, which make the place where etcd stores its data even more reliable than a regular disk (belts *and* ...)"